### ISSUE TITLE
Dropbox: "could not save file to..." Error

### ISSUE THREAD

**Issue by greatlybabled** [koreader/koreader#10714]
* KOReader version: pw2
* Device: kt2 kindle ( 5.12.2.2)

#### Issue
Connected Dropbox to koreader as cloud storage, but when I attempt to download anything it tells me "could not save file to *folder pathname* ". I've tried different file sizes, and different folder locations but I am unable to download any files. I have allowed all permissions in Dropbox. If anyone has a solution for this that would be amazing!

#### Steps to reproduce
- follow these instructions to add Dropbox to koreader 

(https://www.mobileread.com/forums/showthread.phpt=353670#:~:text=In%20KOReader%2C%20I%20went%20to,and%20selected%20Dropbox%20for%20type.)

- tools - cloud storage - Dropbox application - select file - download
- error arises

**Comment by NiLuJe** [comment]
Verbose debug logs (on the latest KOReader versipn) please.

**Comment by greatlybabled** [comment]
[crash.log](https://github.com/koreader/koreader/files/12099308/crash.log)
I think this is right?

**Comment by NiLuJe** [comment]
Yep, except I forgot that the extended logging came *after* the 2023.06.1 release, but, it may be irrelevant, because...

`WARN  DropBoxApi: Download failure: HTTP/1.1 401 Unauthorized`

Which smells like token shenanigans.

**Comment by greatlybabled** [comment]
Hmmm weird , should I regenerate the refresh token?

**Comment by mergen3107** [comment]
Is this you newer short-term token? I believe Dropbox changed the token policy not long time ago. But if you had your long term token from the past, it should still work.

**Comment by hius07** [comment]
If you can see the list of files in the storage, your token is good.

**Comment by cramoisi** [comment]
I'm not sure it's the same issue but I get this message several time on my kobo libra2 when there are some $µù*t in the filename. ; so I just rename the files to get them.

**Comment by Frenzie** [comment]
`*` is an illegal character on Windows, or more to the point FAT32. The others you mentioned should be unproblematic. I wrote http://koreader.rocks/doc/modules/util.html#getSafeFilename for that purpose.

**Comment by Horizen89** [comment]
[crash.log](https://github.com/user-attachments/files/24571766/crash.log)

Any news on this issue?
I followed these instructions: https://www.mobileread.com/forums/showpost.php?p=4319446&postcount=1
I can see all my Dropbox content, but when I try to download files, I get the "Could not save" error message.
I tried different locations and files, renamed event though I don't use special characters... to no avail.

**Comment by hius07** [comment]
From the log:
`Your app (ID: 5787779) is not permitted to access this endpoint because it does not have the required scope 'files.metadata.read'. The owner of the app can enable the scope for the app using the Permissions tab on the App Console. `

It's the Dropbox setting.

**Comment by Horizen89** [comment]
Thanks for the quick response @hius07 ,
However, even after setting all permissions to activated (this was the only one that wasn't, my mistake), the problem still presents itself.

<img width="814" height="223" alt="Image" src="https://github.com/user-attachments/assets/2dfa9f27-1f0b-419f-8a3d-97718063fa6a" />

[crash.log](https://github.com/user-attachments/files/24574552/crash.log)

**Comment by hius07** [comment]
The single related record in the log:

`01/12/26-17:08:25 WARN  DropBoxApi: cannot download file: HTTP/1.1 401 Unauthorized `

**Comment by Horizen89** [comment]
I started from scratch and now it works. My guess is that changing permissions does require a new refresh token, contrary to what the app says.

**Comment by hius07** [comment]
Good to know!

The original issue has been fixed so closing.
